### ISSUE TITLE
COMM UI Library , Beta 1 -- Unable to Join an existing teams call as a teams user, in Calling Sample

### ISSUE THREAD

**Issue by MarteenJL** [Azure/communication-ui-library#4853]
When attempting to join an existing teams call , using a teams meeting link , with a teams identity , in the calling sample, the join call method in CallComposite is not invoked, only "start call"

**What are the steps to reproduce the issue?**
I can repro with my customized Calling sample, unknown if I can with official sample.

**What behavior did you expect?**
Join a call, not start a call.

**In what environment did you see the issue?**

    "@azure/communication-react": "1.18.0-beta.1",
    "@azure/communication-calling": "1.27.1-beta.1 || ^1.26.1",
    "@internal/acs-ui-common": "1.18.0-beta.1",
    "@internal/calling-component-bindings": "1.18.0-beta.1",
    "@internal/calling-stateful-client": "1.18.0-beta.1",
    "@internal/react-components": "1.18.0-beta.1",
    "@internal/react-composites": "1.18.0-beta.1",

- OS & Device: Windows 11 
- Browser: Microsoft Edge
- Browser Version: Version 126.0.2592.87

**Is there any additional information?**

problem is callees is not null/undefined in this scenario ( join a teams call ). In CallComposite.tsx

```
switch (page) {
    case 'configuration':
      pageElement = (
        <ConfigurationPage
          mobileView={props.mobileView}
          startCallHandler={(): void => {
            if (callees) {  **<--- always true**
              adapter.startCall(
                callees,
                /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId
                  ? { alternateCallerId: { phoneNumber: alternateCallerId } }
                  : {}
              );
            } else {
              adapter.joinCall({
                microphoneOn: 'keep',
                cameraOn: 'keep'
              });
            }
          }}
```
here is my workaround ( just an example ):
App.tsx :  
  `const [targetCallees, setTargetCallees] = useState<StartCallIdentifier[] | undefined>([]);`

           if (callDetails.option === 'TeamsIdentity'){

              setTargetCallees(undefined);
            }
[
<img width="452" alt="Screenshot 2024-07-12 102235" src="https://github.com/user-attachments/assets/071fc659-fe9c-4b2f-8b37-852e67b4ec42">
](url)

**Comment by PorterNan** [comment]
Hi,  @MarteenJL!

Thanks for looking into our code base - which function are you using to create adapter? Do you pass in callee there or you are using locator to create the adapter? Would be great to see your code piece of adapter creation, that would helpful to check what is going on from our side

**Comment by MarteenJL** [comment]
Hi @PorterNan  , CallScreen.tsx , slightly modified, is used to create the teams call adapter.  in the useMemo hook
`  const credential = useMemo(() => {

    /* @conditional-compile-remove(teams-identity-support) */
    if (isTeamsIdentityCall) {
      return new AzureCommunicationTokenCredential(token);
    }
    return createAutoRefreshingCredential(toFlatCommunicationIdentifier(userId), token);
  }, [token, userId, /* @conditional-compile-remove(teams-identity-support) */ isTeamsIdentityCall]);

  /* @conditional-compile-remove(teams-identity-support) */
  if (isTeamsIdentityCall) {  **<----------------------------**
    return <TeamsCallScreen afterCreate={afterTeamsCallAdapterCreate} credential={credential}  {...props} />;  
  }`

then a little further down, in the props 
```
type TeamsCallScreenProps = CallScreenProps & {
  afterCreate?: (adapter: TeamsCallAdapter) => Promise<TeamsCallAdapter>;
  credential: AzureCommunicationTokenCredential;
};


/* @conditional-compile-remove(teams-identity-support) */
const TeamsCallScreen = (props: TeamsCallScreenProps): JSX.Element => {

  
  const { afterCreate, callLocator: locator, userId, ...adapterArgs } = props;
  if (!(locator && 'meetingLink' in locator)) {
    throw new Error('A teams meeting locator must be provided for Teams Identity Call.');
  }

  if (!('microsoftTeamsUserId' in userId)) {
    throw new Error('A MicrosoftTeamsUserIdentifier must be provided for Teams Identity Call.');
  }

  
  const teamsAdapterOptions: TeamsAdapterOptions = useMemo(() => {
    return {
      videoBackgroundOptions: {
        videoBackgroundImages
      },
      
    onFetchProfile: async (userId: string, defaultProfile?: Profile): Promise<Profile | undefined> => {
      if (defaultProfile?.displayName) {
        return defaultProfile;
      }
      console.log('userid' + userId); /// this is the caller, with 8:orgid

      if (props.connection)
      {
        let myName = await props.connection.invoke('S..... <snip>

      // You can fetch the display name from GraphAPI or your backend service using userId
        return { displayName: myName }; 
      }
      return { displayName: "NOTKNOWN" }; 
    }
    };
  }, [props]);

  const adapter = useTeamsCallAdapter(
    {
      ...adapterArgs,
      userId,
      locator,
      options: teamsAdapterOptions
    },
    afterCreate,
  );
```
HTH. Maybe its not a bug, just twisting the codebase too much :)

**Comment by MarteenJL** [comment]
check App.tsx, around line 117?             
` setIsTeamsCall(!!callDetails.teamsToken);`
so as the teams token is provided, its a teams call.

**Comment by PorterNan** [comment]
I've opened a PR to fix our sample for this one #4857, I believe it's the same issue in your sample, feel free to apply the same fix to yours

**Comment by prabhjot-msft** [comment]
Closing the issue as it is resolved, @MarteenJL please reopen the issue in case you have any further concerns.
